rest/route/sns.go

```diff
@@ -425,3 +442,69 @@ func (sns *ecsSNS) handleStoppedPod(ctx context.Context, p *model.APIPod, reason
 
 	return nil
 }
+
+func (sns *ecsSNS) cleanupUnrecognizedPod(ctx context.Context, details ecsEventDetail) error {
+	status := ecs.TaskStatus(details.DesiredStatus).ToCocoaStatus()
+	if status == cocoa.StatusStopping || status == cocoa.StatusStopped {
+		// The unrecognized pod is already cleaning up, so no-op.
+		return nil
+	}
+
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		return errors.Wrap(err, "getting service flag for unrecognized pod cleanup")
+	}
+	if flags.UnrecognizedPodCleanupDisabled {
+		return nil
+	}
+
+	// Spot check that the notification is actually for a pod in an
+	// Evergreen-owned cluster.
+	var isInManagedCluster bool
+	for _, c := range sns.env.Settings().Providers.AWS.Pod.ECS.Clusters {
+		if strings.Contains(details.ClusterARN, c.Name) {
+			isInManagedCluster = true
+			break
+		}
+	}
+	if !isInManagedCluster {
+		return nil
+	}
+
+	c, err := sns.makeECSClient(sns.env.Settings())
+	if err != nil {
+		return errors.Wrap(err, "getting ECS client")
+	}
+	defer c.Close(ctx)
+
+	resources := cocoa.NewECSPodResources().
+		SetCluster(details.ClusterARN).
+		SetTaskID(details.TaskARN)
+
+	statusInfo := cocoa.NewECSPodStatusInfo().SetStatus(status)
+
+	podOpts := ecs.NewBasicECSPodOptions().
+		SetClient(c).
+		SetStatusInfo(*statusInfo).
+		SetResources(*resources)
+
+	p, err := ecs.NewBasicECSPod(podOpts)
```

Comment:
This is probably me just misunderstanding but - the documentation for this function says it "initializes a new pod that is backed by ECS".  Later we delete it and its owned resources, but are we just creating a pod here to delete it later? I'm assuming that this pod that we need to clean up should already exist.

Reply:
Gotcha, makes sense. Thanks!